homeassistant/components/modbus/light.py

```diff
@@ -0,0 +1,49 @@
+"""Support for Modbus lights."""
+import logging
+from typing import Optional
+
+from homeassistant.components.light import LightEntity
+from homeassistant.const import CONF_LIGHTS, CONF_NAME
+from homeassistant.helpers.typing import (
+    ConfigType,
+    DiscoveryInfoType,
+    HomeAssistantType,
+)
+
+from .const import CALL_TYPE_COIL, CONF_INPUT_TYPE, MODBUS_DOMAIN
+from .modbus import ModbusHub
+from .switch import ModbusCoilSwitch, ModbusRegisterSwitch
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_platform(
+    hass: HomeAssistantType,
+    config: ConfigType,
+    async_add_entities,
+    discovery_info: Optional[DiscoveryInfoType] = None,
+):
+    """Read configuration and create Modbus light."""
+    if discovery_info is None:
+        _LOGGER.error(
+            "You're trying to init Modbus Light in an unsupported way."
+            " Check the documentation and fix your configuration"
```

Comment:
Very good, but would be excellent if you changed "documentation" to an actual link. Which reminds did you update the doc PR, a lot have happened with the modbus page.

Reply:
Done